distributed/tests/test_semaphore.py

```diff
@@ -0,0 +1,156 @@
+import pickle
+from time import sleep
+
+from distributed import Client, Semaphore, get_client
+from distributed.metrics import time
+from distributed.utils_test import cluster, gen_cluster  # noqa F401
+
+import dask
+from dask.distributed import Client
+
+
+@gen_cluster(client=True)
+async def test_semaphore(c, s, a, b):
+    semaphore = await Semaphore(max_leases=2, name="exasol_db")
```

Comment:
Why the name `exasol_db`?

Reply:
That slipped in for educational purposes, I believe. We should revert this.

Background (other than we're using an Exasol)

We had a discussion about the naming of arguments and where discussing about the `name` argument and whether or not it should be named `resource` since the semaphore is likely used to limit access to a resource

Reply:
I renamed the arguments `id` and `name` to `instance_id` and `resource_name` respectively

Reply:
Note that variables, locks, and queues all use a `name=` parameter.  If this parameter fulfills a similar role then it would be good to keep the name `name` for consistency.  Similarly Client uses `id` for instance id.  We might want to stick to that as well.

Reply:
I'll revert that for consistency then. I still think it wouldn't hurt to refactor that sometime for easier understanding of the code